Makefile

```diff
@@ -62,54 +62,66 @@ TEMP_COMPOSE_YML := $(shell $(if $(IS_WIN), (New-TemporaryFile).FullName, mktemp
 SWARM_HOSTS       = $(shell $(DOCKER) node ls --format="{{.Hostname}}" 2>$(if IS_WIN,null,/dev/null))
 
 
-
-create-stack-file: ## Creates stack file for production as $(output_file) e.g. 'make create-stack-file output_file=stack.yaml'
+.PHONY: config
+create-stack-file: config # TODO: deprecate create-stack-file
+config: ## Creates deploy stack file for production as $(output_file) e.g. 'make config output_file=stack.yaml'
 	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.prod.yml config > $(output_file)
 
+
 .PHONY: build
-build: .env .build-webclient ## Builds all core service images (user `make rebuild` to build w/o cache)
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml build --parallel $(SERVICES_LIST)
+build: .env ## Builds all core service images (user 'make build-nc' to build w/o cache)
+	# Compiling front-end
+	$(MAKE) -C services/web/client compile
+	# Building services
+	export BUILD_TARGET=production; \
+	$(DOCKER_COMPOSE) -f services/docker-compose.build.yml build --parallel
+
+
+.PHONY: rebuild build-nc
+rebuild: build-nc #TODO: deprecate rebuild
+build-nc: .env
+	# Compiling front-end
+	$(MAKE) -C services/web/client clean compile
+	# Building services
+	export BUILD_TARGET=production; \
+	$(DOCKER_COMPOSE) -f services/docker-compose.build.yml build --no-cache --parallel
 
-.PHONY: rebuild
-rebuild: .env .build-webclient
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml build --no-cache --parallel $(SERVICES_LIST)
 
 .PHONY: build-devel
-build-devel: .env .build-webclient ## Builds images of core services for development
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.devel.yml build --parallel
+build-devel: .env ## Builds images of core services for development
+	# Compiling front-end ('compile' target needed since productions stage in Dockerfile of webserver copies client/tools/qooxdoo-kit/build:latest)
+	$(MAKE) -C services/web/client compile compile-dev
+	# Building services
+	export BUILD_TARGET=development; \
```

Comment:
this is a bit weird, so even though the docker-compose.devel.yml is used, one needs to export the BUILD_TARGET?

Reply:
yes I explain that to you better in person .. it is easier, provided the extra changes i have added